File: src/http/task.rs
use crate::dto::task::ScraperTaskQuery;
use crate::model::prelude::ScraperTask;
use crate::model::scraper_task;
use crate::utils::jwt::Claims;
use anyhow::Context;
use sea_orm::{ColumnTrait, DbConn, EntityTrait, QueryFilter};
use spring_sea_orm::pagination::{Pagination, PaginationExt};
use spring_web::axum::Json;
use spring_web::error::Result;
use spring_web::extractor::{Component, Query};
use spring_web::{axum::response::IntoResponse, get};
use spring_web::{delete, patch, post, put};

#[get("/task")]
async fn query_task(
    claims: Claims,
    Query(q): Query<ScraperTaskQuery>,
    Component(db): Component<DbConn>,
    pagination: Pagination,
) -> Result<impl IntoResponse> {
    let mut filter = scraper_task::Column::UserId.eq(claims.uid);
    filter = match q.name {
        Some(name) => filter.and(scraper_task::Column::Name.starts_with(name)),
        None => filter,
    };
    let page = ScraperTask::find()
        .filter(filter)
        .page(&db, pagination)
        .await
        .context("query scraper task failed")?;
    Ok(Json(page))
}

#[post("/task")]
async fn add_task() -> Result<impl IntoResponse> {
    Ok("")
}

#[post("/task/batch")]
async fn add_batch_task() -> Result<impl IntoResponse> {
    Ok("")
}

#[get("/task/:id")]
async fn get_task() -> Result<impl IntoResponse> {
    Ok("")
}

#[delete("/task/:id")]
async fn delete_task() -> Result<impl IntoResponse> {
    Ok("")
}

#[put("/task/:id")]
async fn update_task() -> Result<impl IntoResponse> {
    Ok("")
}

#[get("/task/:id/rule")]
async fn get_task_rule() -> Result<impl IntoResponse> {
    Ok("")
}

#[patch("/task/:id/rule")]
async fn update_task_rule() -> Result<impl IntoResponse> {
    Ok("")
}

#[patch("/task/:id/cron")]
async fn update_task_cron() -> Result<impl IntoResponse> {
    Ok("")
}

#[get("/task/:id/schedule")]
async fn get_task_schedule_info() -> Result<impl IntoResponse> {
    Ok("")
}

#[patch("/task/:id/schedule")]
async fn update_task_schedule_info() -> Result<impl IntoResponse> {
    Ok("")
}

File: cloud/src/api/task.ts
import { Graph } from "types/NodeTypes";
import { Page } from "../types/Page";
import { ScheduleConfig, ScraperTaskData, Task } from "../types/Task";
import ajax from "../utils/ajax";

const task = () => ajax("/task");

/** 获取任务详情（含 `rule`、`data`，与后端 `GET /task/{id}` 一致） */
export async function getTask(id: string): Promise<Task> {
  return task().path(id).get() as Promise<Task>;
}

/** 更新任务（须同时传 `rule`，避免覆盖为空；`data` 与后端 `ScraperTaskData` 信封一致） */
export async function updateTask(
  id: string,
  body: { data?: ScraperTaskData | null; rule: unknown }
): Promise<Task> {
  return task().path(id).payload(body).put() as Promise<Task>;
}

export async function queryTask(
  name: string,
  page: number,
  size: number,
  startTime?: string,
  endTime?: string
) {
  return task()
    .query({ name, page, size, startTime, endTime })
    .get() as Promise<Page<Task>>;
}

export async function addTask(partialTask: Partial<Task>): Promise<Task> {
  return task().payload(partialTask).post() as Promise<Task>;
}

export async function getRule(taskId: string) {
  return task().path(taskId).path("/rule").get() as Promise<Graph>;
}

export async function renameTask(id: string, name: string) {
  return task().path(id).payload({ name }).put() as Promise<Task>;
}

export async function deleteTask(id: string) {
  return task().path(id).delete();
}

export async function updateRemoteCron(id: string, cron: string) {
  return task().path(id).path("cron").payload(cron).patch() as Promise<Task>;
}

export async function getScheduleConfig(id: string) {
  return task()
    .path(id)
    .path("schedule")
    .get() as Promise<ScheduleConfig>;
}

export async function saveScheduleConfig(id: string, config: ScheduleConfig) {
  return task()
    .path(id)
    .path("schedule")
    .payload(config)
    .patch();
}
